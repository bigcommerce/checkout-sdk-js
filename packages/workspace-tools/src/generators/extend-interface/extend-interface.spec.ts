import path from 'path';

import extendInterface from './extend-interface';

describe('extendInterface', () => {
    it('extends interface with other matching interfaces', async () => {
        const options = {
            inputPath: path.join(__dirname, '/__fixtures__/**/index.ts'),
            outputPath: path.join(__dirname, '/__temp__/output.ts'),
            outputMemberName: 'ExtendedInterface',
            memberPattern: '^Interface.$',
            targetPath: path.join(__dirname, '/__fixtures__/foobar-interface/index.ts'),
            targetMemberName: 'FoobarInterface',
        };

        expect(await extendInterface(options)).toMatchSnapshot();
    });

    it('handles scenario where no matching interface is found', async () => {
        const options = {
            inputPath: path.join(__dirname, '/__fixtures__/**/index.ts'),
            outputPath: path.join(__dirname, '/__temp__/output.ts'),
            outputMemberName: 'ExtendedDummyInterface',
            memberPattern: '^DummyInterface.$',
            targetPath: path.join(__dirname, '/__fixtures__/foobar-interface/index.ts'),
            targetMemberName: 'FoobarInterface',
        };

        expect(await extendInterface(options)).toMatchSnapshot();
    });
});
