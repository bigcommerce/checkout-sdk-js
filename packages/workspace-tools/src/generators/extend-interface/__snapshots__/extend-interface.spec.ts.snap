// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extendInterface extends interface with other matching interfaces 1`] = `
"import { FoobarInterface } from '../__fixtures__/foobar-interface';
import { InterfaceA } from '../__fixtures__/interface-a';
import { InterfaceB } from '../__fixtures__/interface-b';
export type ExtendedInterface = FoobarInterface & InterfaceA & InterfaceB;
"
`;

exports[`extendInterface handles scenario where no matching interface is found 1`] = `
"import { FoobarInterface } from '../__fixtures__/foobar-interface';
export type ExtendedDummyInterface = FoobarInterface;
"
`;
